packages/core/demo/data/bar.ts

```diff
@@ -306,16 +307,53 @@ export const stackedBarTimeSeriesData = {
 	]
 };
 
+// export const stackedBarTimeSeriesOptions = {
+// 	title: "Stacked bar (time series)",
+// 	axes: {
+// 		left: {
+// 			primary: true,
+// 			stacked: true
+// 		},
+// 		bottom: {
+// 			scaleType: "time",
+// 			secondary: true
+// 		}
+// 	}
+// };
+
+// export const stackedBarTimeSeriesData = [
+// 	{ group: "Dataset 1", date: new Date(2019, 0, 1), value: 10000 },
+// 	{ group: "Dataset 1", date: new Date(2019, 0, 5), value: 65000 },
+// 	{ group: "Dataset 1", date: new Date(2019, 0, 8), value: 10000 },
+// 	{ group: "Dataset 1", date: new Date(2019, 0, 13), value: 49213 },
+// 	{ group: "Dataset 1", date: new Date(2019, 0, 17), value: 51213 },
+// 	{ group: "Dataset 2", date: new Date(2019, 0, 3), value: 75000 },
+// 	{ group: "Dataset 2", date: new Date(2019, 0, 6), value: 57312 },
+// 	{ group: "Dataset 2", date: new Date(2019, 0, 8), value: 21432 },
+// 	{ group: "Dataset 2", date: new Date(2019, 0, 15), value: 70323 },
+// 	{ group: "Dataset 2", date: new Date(2019, 0, 19), value: 21300 },
+// 	{ group: "Dataset 3", date: new Date(2019, 0, 1), value: 50000 },
+// 	{ group: "Dataset 3", date: new Date(2019, 0, 5), value: 15000 },
+// 	{ group: "Dataset 3", date: new Date(2019, 0, 8), value: 20000 },
+// 	{ group: "Dataset 3", date: new Date(2019, 0, 13), value: 39213 },
+// 	{ group: "Dataset 3", date: new Date(2019, 0, 17), value: 61213 },
+// 	{ group: "Dataset 4", date: new Date(2019, 0, 2), value: 10 },
+// 	{ group: "Dataset 4", date: new Date(2019, 0, 6), value: 37312 },
+// 	{ group: "Dataset 4", date: new Date(2019, 0, 8), value: 51432 },
+// 	{ group: "Dataset 4", date: new Date(2019, 0, 15), value: 40323 },
+// 	{ group: "Dataset 4", date: new Date(2019, 0, 19), value: 31300 },
+// ];
+
 export const stackedBarTimeSeriesOptions = {
 	title: "Stacked bar (time series)",
 	axes: {
 		left: {
-			primary: true,
+			identifier: "value",
```

Comment:
`identifier` is awkward, and kinda unexpected ... As a user I would expect to use a `key` or `id` property instead of `identifier` ... any reason we'd want to stick with `identifier` over something else?

Reply:
I added a comment here yesterday which I don't see anymore. I'll post again.

I'd like to involve @natashadecoste @zvonimirfras @caesarsol in the discussion here.

We could really choose any name we'd like instead of `identifier`. The reason I went with `identifier` is because `key` & `id` would show up in different places in the library codebase and could cause confusion in the code.

Also I felt that `key` & `id` are more likely to show up in user data as well, where `identifier` seems a bit unlikely to show up there.

Reply:
I'd still argue `key: "key"` or `id: "id"` is more readable vs. `identifier: "id"` ... Internally we can use whatever, `getIdentifier` or `getDimensionKey` or whatever is absolutely appropriate there ... The options on the user side are pretty discoverable right now, so I'd hate to break that with an awkward value selector

Reply:
I would still propose something along these lines: https://github.com/carbon-design-system/carbon-charts/issues/516#issuecomment-591047052

`map` still isn't totally ideal, but perhaps it's getting closer? Basically something that makes it clear that the relationship is a mapping between the overall dataset and the specific axis.

Reply:
The `map` function call is a bit much IMO. handling calling that function every where we need a value is going to be too complicated.

I think if you were intending the values of an axis to be `a + b` mapped through every datapoint, you should just provide `c = a + b` in every datapoint manually

Reply:
I was initially ok with `identifier`. I'd find `id` a bit misleading.
`key` could make sense.
`propName` possibly
`mapsTo` possibly too

Actually `mapsTo` is becoming my favorite 😸

Reply:
I'm down for `mapsTo` 👍 Bonus points because extending to be a function later would be trivial ... I will note that:

> handling calling that function every where we need a value is going to be too complicated.

Shouldn't be the case ... in the event that we support a function for `mapsTo`, we should have one (and only one) place that we retrieve data internally - that location can handle calling `mapsTo` (and any other functions we may need) ... generally pre-processing the data to add the appropriate columns should be preferred, but I don't think it's unreasonable to design with the eventuality of `string | (any) => any` in mind

Reply:
I would even argue that _today_ we should have centralized data sourcing, but that's a different conversation 👍

Reply:
- In data science these are often called "features", but I doubt `feature` would be a clear name for users of any background.
 - In JS this is of course an object `key`, but I think this is more of an implementation detail than a logic step;
   - If we look at similar concepts used by D3, pretty much a standard in this world, they often use the term `accessor`: https://github.com/d3/d3-shape#line_x
   - Another standard worth mentioning: Lodash has the `property` function which accesses an object/array property by string: https://lodash.com/docs/#property
 - If we look at it from the point of view of the data, being it tabular would make this a `column`. This would also make clear that the data is thought as a table, and it would be decoupled from the actual array-of-objects representation.

I guess we need to define if we want to only support strings, or add functions as well as @cal-smith proposes. @theiliad is actually making a good point saying that a simple `.map(d => ({ ...d, c: d.a + d.b }))` on the dataset to add a new "column" is pretty easy to do for the user.

If we choose only string properties, I'd go with `column`.
Otherwise I think `accessor` is pretty explicit to support both strings and functions.

Reply:
Great points @caesarsol 👍 Definitly agree that `feature` or `dimension` would be unintuitive for most users ... `column`, `accessor`, and `mapsTo` are definitely my favorite options right now ... My preference leans toward `column` or `mapsTo`, just because `accessor` isn't the usual term outside of the D3 space - it's a reasonable option, but not my first choice.

Reply:
I think my preference is towards `mapsTo` or `dimension`